server/events/project_command_runner.go

```diff
@@ -152,6 +153,31 @@ func (p *ProjectOutputWrapper) updateProjectPRStatus(commandName models.CommandN
 	return result
 }
 
+type FeatureAwareProjectCommandRunner struct {
+	ProjectCommandRunner
+	FeatureAllocator feature.Allocator
+}
+
+func (f *FeatureAwareProjectCommandRunner) Apply(ctx models.ProjectCommandContext) models.ProjectResult {
```

Comment:
Can you add some  tests for this decorator as well?

Reply:
Yeah totally just added some.